### PR TITLE
refactor(client): replace whitespaces in tokenSetCacheKey

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

- Replace whitespaces in `tokenSetCacheKey` via `encodeURI` method
    - Fix related unit tests.

## Testing

Pass all existing unit tests.

### CODE REVIEW

File: packages/client/src/index.ts [hunk 1]
```diff
@@ -210,7 +210,9 @@ export default class LogtoClient {
   }
 
   private get tokenSetCacheKey() {
-    return `${TOKEN_SET_CACHE_KEY}::${this.oidcConfiguration.issuer}::${this.clientId}::${this.scope}`;
+    return encodeURI(
+      `${TOKEN_SET_CACHE_KEY}::${this.oidcConfiguration.issuer}::${this.clientId}::${this.scope}`
```
Comment (xiaoyijun) [inline packages/client/src/index.ts:214]: we just need to encode `scope` ?

File: packages/client/src/index.test.ts [hunk 2]
```diff
@@ -24,7 +24,9 @@ const CLIENT_ID = 'client1';
 const SUBJECT = 'subject1';
 const REDIRECT_URI = 'http://localhost:3000';
 const CODE = 'code1';
-const LOGTO_TOKEN_SET_CACHE_KEY = `LOGTO_TOKEN_SET_CACHE::${ISSUER}::${CLIENT_ID}::${DEFAULT_SCOPE_STRING}`;
+const LOGTO_TOKEN_SET_CACHE_KEY = encodeURI(
```
Comment (wangsijie) [inline packages/client/src/index.test.ts:27]: `encodeURIComponent` would be more safe?

**Review by wangsijie — CHANGES_REQUESTED** [review]

File: packages/client/src/index.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/client/src/index.ts:214]: After disscussing with sijie, encodeURIComponent the whole key will be safer

Comment (IceHe) [reply packages/client/src/index.ts:214]: Fixed.

References:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

**Review by wangsijie — APPROVED** [review]

**Pull request merged**